fuse/data/tokenizers/modular_tokenizer/inject_utils.py

```diff
@@ -155,6 +166,9 @@ def build_scalars(
         # for each element, whether it's a scalar or not
         all_scalars_valid_mask = []
         scalar_default_unfound_value = -1000.0
+        external_embeddings_info = dict()  # a dict mapping location -> embedding input
+        num_tokens_token_so_far = 0
```

Comment:
I was expecting for num_tokens_token_so_far to be updated also when adding scalars.
Plus, it's better if this variable will be used to also build  all_scalars_values and all_scalars_valid_mask

Reply:
done